### DISCUSSION TITLE
Plot tidak muncul dan menu sidebar tidak terbuka di dashboard

### DISCUSSION THREAD

**Discussion by AngelMartha** [teamalgoritma/community#683] [Q&A] [answered]
[capstone_DV.zip](https://github.com/teamalgoritma/community/files/11652004/capstone_DV.zip)

Ketika tampilan dashboard sudah muncul, pada menu sidebar umur dan pendidikan tidak bisa terbuka serta di sidebar gender dan daerah tidak muncul tampilan plotnya.

**Comment by jafarhf1** [comment, marked as answer]
Dear @AngelMartha,

Mengenai menu sidebar umur dan pendidikan yang tidak muncul pada dashboardnya, ada ketidaksesuaian parameter `tabName` pada bagian **ui.R** yang ada di `menuItem()` dan `tabItem()`. Misalnya adalah pada bagian menu item yang kedua tertulis tabName=''Golongan Umur" di `menuItem()`  tetapi pada bagian `tabItem()` hanya tertulis tabName="Umur" sehingga menjadi tidak sesuai. Selain itu, penggunaan nama pada `tabName`  sebaiknya tidak menggunakan spasi melainkan "_" untuk memisahkan dua kata, hal ini juga bisa menyebabkan sidebarnya tidak muncul.  Adapun mengenai plot dan tabel yang tidak muncul hal ini disebabkan ada kesalahan penggunaan fungsi yaitu seharusnya `dataTableOutput()` bukan tableOutput(). Selain itu, untuk menampilkan pie chart tidak bisa menggunakan `ggplotly()`, alternatifnya adalah menggunakan library `plotly` nya langsung atau menggunakan `echart4`

Demikian yang dapat kami sampaikan. Semoga membantu.

**Comment by AngelMartha** [comment]
Baik, terima kasih banyak pak @jafarhf1 . Masalah sidebar umur dan pendidikan sudah teratasi dan plot pada keduanya juga sudah muncul. Begitu juga pada table yang sudah terlihat. Namun, untuk pie chart masih belum bisa teratasi. Berikut saya izin melampirkan sintaks untuk pie chartnya pak 
**> plotly** 
**Di bagian server** 
output$plot_daerah <- renderPlotly({
  plot_ly(agg4, x = 0, y = 0, type = "scatter", mode = "markers", marker = list(size = 2 * max(agg4$sum_orang4), symbol = "circle", color = agg4$daerah)) %>%
    layout(title = NULL, showlegend = FALSE, xaxis = list(showgrid = FALSE, zeroline = FALSE, showticklabels = FALSE), 
           yaxis = list(showgrid = FALSE, zeroline = FALSE, showticklabels = FALSE))
})
**Di bagian UI** 
fluidRow(
  box(
    title = "Berdasarkan Daerah pada Tahun 2011-2022",
    status = "success",
    solidHeader = TRUE,
    width = 6,
    plotlyOutput(outputId = "plot_daerah")
  )
)

**>echarts4r**
Di bagian server
output$plot_daerah <- renderEcharts4r({
  agg4 %>%
    e_charts() %>%
    e_scatter(x = 0, y = 0, symbol = "circle", color = ~daerah, data = .) %>%
    e_visual_map(max = 2 * max(agg4$sum_orang4)) %>%
    e_tooltip(formatter = "{b}") %>%
    e_title(NULL, NULL, NULL)
})
**>Di bagian UI**
fluidRow(
  box(
    title = "Berdasarkan Daerah pada Tahun 2011-2022",
    status = "success",
    solidHeader = TRUE,
    width = 6,
    echarts4rOutput(outputId = "plot_daerah")
  )
)
Saya telah mencoba run keduanya namun pie chart tetap tidak terlihat. Apakah terdapat kesalahan sintaks pak?
Terima kasih pak

**Reply by jafarhf1** [reply]
Dear @AngelMartha ,

Apakah ada tulisan error yang muncul di dashboard atau di console Rstudionya?
